From 68944497415ba8aabed07ffab03258c56eb165cd Mon Sep 17 00:00:00 2001
From: Dan Lambright <dlambrig@redhat.com>
Date: Tue, 15 Dec 2015 19:30:45 -0500
Subject: [PATCH 197/214] cluster/tier: fix tier-max-files bookeeping and help

The option tier-max-files represents the maximum number
of files trasferred by a node in a gives cycle. Fix help message
to reflect the "per node" aspect. The code transferred one
more file per cycle than the given value.

Also change the default values of max file and max bytes to
very large values, effectively we will not throttle migration
unless the administrator requests it via CLI.

This is a backport of 12984.

> Change-Id: Ic2949ed3d8c35afe7c9ae4db72195603cfb2e28f
> BUG: 1292671
> Signed-off-by: Dan Lambright <dlambrig@redhat.com>
> Reviewed-on: http://review.gluster.org/12984
> Tested-by: NetBSD Build System <jenkins@build.gluster.org>
> Tested-by: Gluster Build System <jenkins@build.gluster.com>
Signed-off-by: Dan Lambright <dlambrig@redhat.com>

Change-Id: I516cc7a641ae82d9686fc797c606ab1a996db191
BUG: 1291152
Reviewed-on: https://code.engineering.redhat.com/gerrit/64229
Reviewed-by: Joseph Fernandes <josferna@redhat.com>
Tested-by: Joseph Fernandes <josferna@redhat.com>
---
 xlators/cluster/dht/src/dht-rebalance.c         |    1 -
 xlators/cluster/dht/src/tier.c                  |    2 +-
 xlators/mgmt/glusterd/src/glusterd-volume-set.c |    8 ++++----
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/xlators/cluster/dht/src/dht-rebalance.c b/xlators/cluster/dht/src/dht-rebalance.c
index 0c6594b..31c61b0 100644
--- a/xlators/cluster/dht/src/dht-rebalance.c
+++ b/xlators/cluster/dht/src/dht-rebalance.c
@@ -3632,7 +3632,6 @@ gf_defrag_pause_tier (xlator_t *this, gf_defrag_info_t *defrag)
                     (defrag->defrag_status != GF_DEFRAG_STATUS_STARTED)) {
                         goto out;
                 }
-
                 usleep (usec_sleep);
         }
 
diff --git a/xlators/cluster/dht/src/tier.c b/xlators/cluster/dht/src/tier.c
index 95a4960..73e39e5 100644
--- a/xlators/cluster/dht/src/tier.c
+++ b/xlators/cluster/dht/src/tier.c
@@ -531,7 +531,7 @@ abort:
                                 xdata_response = NULL;
                         }
 
-                        if ((total_files > defrag->tier_conf.max_migrate_files)
+                        if ((total_files >= defrag->tier_conf.max_migrate_files)
                             || (total_migrated_bytes >
                                 defrag->tier_conf.max_migrate_bytes)) {
                                 gf_msg (this->name, GF_LOG_INFO, 0,
diff --git a/xlators/mgmt/glusterd/src/glusterd-volume-set.c b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
index 71af98a..8b35225 100644
--- a/xlators/mgmt/glusterd/src/glusterd-volume-set.c
+++ b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
@@ -2294,22 +2294,22 @@ struct volopt_map_entry glusterd_volopt_map[] = {
         { .key         = "cluster.tier-max-mb",
           .voltype     = "cluster/tier",
           .option      = "tier-max-mb",
-          .value       = "1000",
+          .value       = "10000",
           .op_version  = GD_OP_VERSION_3_7_6,
           .flags       = OPT_FLAG_CLIENT_OPT,
           .validate_fn = validate_tier,
           .description = "The maximum number of MB that may be migrated"
-          " in any direction in a given cycle."
+          " in any direction in a given cycle by a single node."
         },
         { .key         = "cluster.tier-max-files",
           .voltype     = "cluster/tier",
           .option      = "tier-max-files",
-          .value       = "5000",
+          .value       = "50000",
           .op_version  = GD_OP_VERSION_3_7_6,
           .flags       = OPT_FLAG_CLIENT_OPT,
           .validate_fn = validate_tier,
           .description = "The maximum number of files that may be migrated"
-          " in any direction in a given cycle."
+          " in any direction in a given cycle by a single node."
         },
         { .key         = "features.ctr-enabled",
           .voltype     = "features/changetimerecorder",
-- 
1.7.1

